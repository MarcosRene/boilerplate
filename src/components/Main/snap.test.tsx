import { render, screen } from '@testing-library/react'

import { Main } from '.'

describe('<Main />', () => {
  it('should render the heading', () => {
    const { container } = render(<Main />)

    const title = screen.getByRole('heading', { name: /react avanÃ§ado/i })
    const description = screen.getByRole('heading', {
      name: /typescript, reactJS, nextJS e styled components/i
    })

    expect(title).toBeInTheDocument()
    expect(description).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render the colors correctly', () => {
    const { container } = render(<Main />)

    expect(container.firstChild).toHaveStyle({
      'background-color': 'var(--bg-color)'
    })
  })
})
