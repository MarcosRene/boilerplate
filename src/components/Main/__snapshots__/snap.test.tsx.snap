// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> should render the heading 1`] = `
.c0 {
  height: 100%;
  width: 100%;
  padding: 3rem;
  color: var(--color);
  background-color: var(--bg-color);
  text-align: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c1 {
  width: 25rem;
  margin-bottom: 2rem;
}

.c2 {
  font-size: 2.5rem;
}

.c3 {
  font-size: 2rem;
  font-weight: 400;
}

.c4 {
  margin-top: 3rem;
  width: min(30rem,100%);
}

<main
  class="c0"
>
  <img
    alt="Imagem de um átomo com um texto React Avançado do lado direito."
    class="c1"
    src="/logo.svg"
  />
  <h1
    class="c2"
  >
    React Avançado
  </h1>
  <h2
    class="c3"
  >
    Typescript, ReactJS, NextJS e Styled Components
  </h2>
  <img
    alt="Homem sentado em uma cadeira de frente para três quadros. Um quadro verde grande e no centro e dois quadros brancos e pequenos aos lados do quadro grande."
    class="c4"
    src="/hero-illustration.svg"
  />
</main>
`;
